lib/Doctrine/ORM/Query/ParameterTypeInferer.php

```diff
@@ -44,6 +44,9 @@ class ParameterTypeInferer
      * - Type (\Doctrine\DBAL\Types\Type::*)
      * - Connection (\Doctrine\DBAL\Connection::PARAM_*)
      *
+     * @deprecated Infer Type will be removed in the future for setParameter method,
```

Comment:
Will this not cause IDEs and static analyzers to mark _all_ calls to this method as deprecated?

Reply:
yes that's correct but isn't this the desired outcome of the opened issue or did I misunderstood the issue?

Reply:
I think you misunderstood. IMO what's needed here is some usage of `doctrine/deprecations` in the cases that should be deprecated and only those.

Reply:
My bad then. Does this mean, if I understood it correctly now, that we just need to trigger a deprecation warning for the third argument of `setParameter` method by using `doctrine/deprecations`?

Reply:
Yes, I believe that's indeed what's needed 👍

Reply:
thanks for the clarification, I'll update the PR later today

Reply:
Do I have to update the existing tests or create a new one? How is this usually handled?

Reply:
It looks like you had the former anyway, but it might be good to do the latter too, to demonstrate that it affects all classes extending `AbstractQuery`, and not just `NativeQuery`

Reply:
I found only two classes that extend the `AbstractQuery` and the tests for both classes are updated. Can you think about any use-case for which we might need new tests for this?

Reply:
Since `AbstractQuery` is neither final nor internal, anybody can extend it outside `doctrine/orm` if they want to, and they should get the deprecation too.